### PR TITLE
New kstat module zfs_metaslab_stats

### PR DESCRIPTION
**Pull request by datacore-yash**
This will be useful for tracking metaslab thrashing
Signed by: Yash Jain <yash.jain@datacore.com>

### CODE REVIEW

File: ZFSin/zfs/include/sys/metaslab_impl.h [hunk 1]
```diff
@@ -433,7 +440,7 @@ struct metaslab {
 	 */
 	uint64_t	ms_selected_txg;
 
-	uint64_t	ms_alloc_txg;	/* last successful alloc (debug only) */
+	uint64_t    ms_alloc_txg;   /* last successful alloc (debug only) */
```
Comment (aviksildc) [inline ZFSin/zfs/include/sys/metaslab_impl.h:443]: [Nitpick] unnecessary change

File: ZFSin/zfs/lib/libzpool/kernel.c [hunk 2]
```diff
@@ -254,6 +254,7 @@ void
 kstat_named_init(kstat_named_t *knp, const char *name, uchar_t type)
 {}
 
+
```
Comment (aviksildc) [inline ZFSin/zfs/lib/libzpool/kernel.c:257]: [Nitpick] not required

File: ZFSin/zfs/module/zfs/metaslab.c [hunk 3]
```diff
@@ -2892,17 +2936,17 @@ metaslab_sync_done(metaslab_t *msp, uint64_t txg)
...
 	    msp->ms_selected_txg + metaslab_unload_delay < txg) {
-		for (int t = 1; t < TXG_CONCURRENT_STATES; t++) {
-			VERIFY0(range_tree_space(
-			    msp->ms_allocating[(txg + t) & TXG_MASK]));
-		}
-		if (msp->ms_allocator != -1) {
-			metaslab_passivate(msp, msp->ms_weight &
-			    ~METASLAB_ACTIVE_MASK);
-		}
+			for (int t = 1; t < TXG_CONCURRENT_STATES; t++) {
```
Comment (aviksildc) [inline ZFSin/zfs/module/zfs/metaslab.c:2939]: [Nitpick] indentation changes; not required

**Review by aviksildc — APPROVED** [review]
LGTM other than few nitpicks

File: ZFSin/zfs/module/zfs/metaslab.c [hunk 4]
```diff
@@ -1610,6 +1639,12 @@ metaslab_load(metaslab_t *msp)
 	if (msp->ms_loaded)
 		return (0);
 	VERIFY(!msp->ms_loading);
+	ksp = msp->ms_group->mg_kstat;
+	if (ksp != NULL) {
```
Comment (aviksildc) [inline ZFSin/zfs/module/zfs/metaslab.c:1643]: Do you need to check for `if (msp->ms_group != NULL)` before this, similar to what you did for `metaslab_unload()`?

Comment (datacore-yash) [reply ZFSin/zfs/module/zfs/metaslab.c]: added null check
